### PR TITLE
add: PGN 127501 binary switch bank status

### PR DESCRIPTION
**Pull request by sbender9**

### CODE REVIEW

File: n2kMapper.js [hunk 1]
```diff
@@ -65,45 +68,49 @@ function getValue (n2k, theMapping, state) {
...
+            return (
+              typeof theMapping.filter === 'undefined' ||
+              theMapping.filter(n2k, state)
+            )
+          } catch (ex) {
+            process.stderr.write(ex + ' ' + n2k)
+            return false
+          }
+        })
+        .map(function (theMapping) {
```
Comment (tkurki) [inline n2kMapper.js]: Now the code assumes that there is only one mapping if the mapping is a function.

If the test for function was down here you could mix traditional mappings and functions, theMappings would always be an array and the code would have branching in just one spot.

**Comment by tkurki** [comment]
Your comments?

**Comment by sbender9** [comment]
Totally forgot about this. I'll work on it tonight...

**Comment by sbender9** [comment]
@tkurki this should be good now. any issues?

File: n2kMapper.js [hunk 2]
```diff
@@ -81,30 +84,36 @@ var toValuesArray = function (theMappings, n2k, state) {
...
-              : theMapping.node
-          var value =
-            typeof theMapping.source === 'function'
-              ? theMapping.source(n2k, state)
-              : getValue(n2k, theMapping, state)
-          var allowNull =
-            typeof theMapping.allowNull !== 'undefined' &&
+          if (typeof theMapping === 'function') {
+            var mapped = theMapping(n2k, state)
+            updates.push.apply(updates, mapped)
```
Comment (tkurki) [inline n2kMapper.js]: Why not just `updates.push(theMapping(n2k, state))`?

File: n2kMapper.js [hunk 3]
```diff
@@ -81,30 +84,36 @@ var toValuesArray = function (theMappings, n2k, state) {
           return false
         }
       })
-      .map(function (theMapping) {
+      .reduce(function (updates, theMapping) {
```
Comment (tkurki) [inline n2kMapper.js]: Could the reducer be a static function so that we would not need to create one for each incoming pgn?

**Comment by sbender9** [comment]
I can't seem to get these conflicts resolved... recreating....

**Pull request closed without merging**